yesod-persistent/Yesod/Persist/Core.hs

```diff
@@ -163,7 +165,30 @@ getBy404 key = do
         Nothing -> notFound'
         Just res -> return res
 
+-- | Create a new record in the database, returning an automatically
+-- created key, or raise a 400 bad request if a uniqueness constraint
+-- is violated.
+insert400 :: (MonadIO m, PersistUniqueWrite backend, PersistRecordBackend record backend)
+          => record
+          -> ReaderT backend m (Key record)
```

Comment:
Looks like the lack of CPP code as used in `getBy404` would require a `persistent >= 2.5` in the .cabal file. Ideally we'd keep support for older persistent versions. Is doing so a problem?